flow-typed/custom/jsdom.js

```diff
@@ -0,0 +1,320 @@
+    HTMLHeadingElement: typeof HTMLHeadingElement,
+    HTMLHeadElement: typeof HTMLHeadElement,
+    HTMLHRElement: typeof HTMLHRElement,
+    HTMLHtmlElement: typeof HTMLHtmlElement,
+    HTMLIFrameElement: typeof HTMLIFrameElement,
+    HTMLImageElement: typeof HTMLImageElement,
+    HTMLInputElement: typeof HTMLInputElement,
+    HTMLLabelElement: typeof HTMLLabelElement,
+    HTMLLegendElement: typeof HTMLLegendElement,
+    HTMLLIElement: typeof HTMLLIElement,
+    HTMLLinkElement: typeof HTMLLinkElement,
+    HTMLMapElement: typeof HTMLMapElement,
+    HTMLMarqueeElement: typeof HTMLMarqueeElement,
+    HTMLMediaElement: typeof HTMLMediaElement,
+    HTMLMenuElement: typeof HTMLMenuElement,
+    HTMLMetaElement: typeof HTMLMetaElement,
+    HTMLMeterElement: typeof HTMLMeterElement,
+    HTMLModElement: typeof HTMLModElement,
+    HTMLObjectElement: typeof HTMLObjectElement,
+    HTMLOListElement: typeof HTMLOListElement,
+    HTMLOptGroupElement: typeof HTMLOptGroupElement,
+    HTMLOptionElement: typeof HTMLOptionElement,
+    HTMLOutputElement: typeof HTMLOutputElement,
+    HTMLParagraphElement: typeof HTMLParagraphElement,
+    HTMLParamElement: typeof HTMLParamElement,
+    HTMLPictureElement: typeof HTMLPictureElement,
+    HTMLPreElement: typeof HTMLPreElement,
+    HTMLProgressElement: typeof HTMLProgressElement,
+    HTMLQuoteElement: typeof HTMLQuoteElement,
+    HTMLScriptElement: typeof HTMLScriptElement,
+    HTMLSelectElement: typeof HTMLSelectElement,
+    HTMLSourceElement: typeof HTMLSourceElement,
+    HTMLSpanElement: typeof HTMLSpanElement,
+    HTMLStyleElement: typeof HTMLStyleElement,
+    HTMLTableCaptionElement: typeof HTMLTableCaptionElement,
+    HTMLTableCellElement: typeof HTMLTableCellElement,
+    HTMLTableColElement: typeof HTMLTableColElement,
+    HTMLTableElement: typeof HTMLTableElement,
+    HTMLTimeElement: typeof HTMLTimeElement,
+    HTMLTitleElement: typeof HTMLTitleElement,
+    HTMLTableRowElement: typeof HTMLTableRowElement,
+    HTMLTableSectionElement: typeof HTMLTableSectionElement,
+    HTMLTemplateElement: typeof HTMLTemplateElement,
+    HTMLTextAreaElement: typeof HTMLTextAreaElement,
+    HTMLTrackElement: typeof HTMLTrackElement,
+    HTMLUListElement: typeof HTMLUListElement,
+    HTMLUnknownElement: typeof HTMLUnknownElement,
+    HTMLVideoElement: typeof HTMLVideoElement,
+    StyleSheet: typeof StyleSheet,
+    MediaList: typeof MediaList,
+    CSSStyleSheet: typeof CSSStyleSheet,
+    CSSRule: typeof CSSRule,
+    CSSStyleRule: typeof CSSStyleRule,
+    CSSMediaRule: typeof CSSMediaRule,
+    CSSImportRule: typeof CSSImportRule,
+    CSSStyleDeclaration: typeof CSSStyleDeclaration,
+    StyleSheetList: typeof StyleSheetList,
+    XPathExpression: typeof XPathExpression,
+    XPathResult: typeof XPathResult,
+    XPathEvaluator: typeof XPathEvaluator,
+    NodeFilter: typeof NodeFilter,
+    close: () => void;
+    [key: string]: mixed,
```

Comment:
These last two; `close` and the indexer, they are added by me. `close` is part of the JSDOM API and the ability to attach random stuff to `window` is used often in our code as it stands, so the indexer permits that.